### ISSUE TITLE
Pylance not detecting invalid type passed to a method

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pyright#3951]
## Environment data

-   Language Server version: Pylance insiders 2022.9.21
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.3

## Code Snippet

```python
T2 = TypeVar('T2')

class MyClass(Generic[T2]):
    def meth_1(self, x: T2) -> T2: ...  # T here
    def meth_2(self, x: T2) -> T2: ...  # and here are always the same

m = MyClass()  # type: MyClass[int]
m.meth_1(1)    # OK
m.meth_2('a')  # This should be an error
```

## Repro Steps

On the provided code, mypy outputs the following (and https://peps.python.org/pep-0484/#scoping-rules-for-type-variables says it should be an error)

```
PS C:\Users\aku91> mypy test.py
test.py:37: error: Argument 1 to "meth_2" of "MyClass" has incompatible type "str"; expected "int"
Found 1 error in 1 file (checked 1 source file)
```

**Comment by erictraut** [comment]
This appears to be working correctly. If you enable type checking diagnostics, you should see the same error that mypy reports.

![Screen Shot 2022-09-15 at 10 21 13 AM](https://user-images.githubusercontent.com/7040122/190469851-eac83858-28e2-4c65-a779-1e794780987a.png)

**Comment by rchiodo** [comment]
Sorry didn't realize there was a setting. My other issue might be working too.

**Comment by rchiodo** [comment]
The appropriate setting is this:

```json
    "python.analysis.typeCheckingMode": "basic"
```
